plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java

```diff
@@ -257,4 +271,19 @@ public void validate()
                 !countDistinctPushdownEnabled || aggregationPushdownEnabled,
                 "Invalid configuration: pinot.aggregation-pushdown.enabled must be enabled if pinot.count-distinct-pushdown.enabled");
     }
+
+    @AssertTrue(message = "Controller URL should use HTTPS scheme when TLS is enabled for Pinot connector")
+    public boolean isControllerUrlValid()
+    {
+        if (!isTlsEnabled()) {
+            return true;
+        }
+        for (URI url : getControllerUrls()) {
+            String scheme = url.getScheme();
+            if (scheme != null && !"https".equalsIgnoreCase(scheme)) {
+                return false;
```

Comment:
When a scheme is omitted in `pinot.controller-urls` & `pinot.tls.enabled` is enabled, users will expect using https. This should be fixed or improve the error message (or document).

Reply:
Right, the error msg will show that when tls is enabled, the scheme should be https.

Reply:
I separated the check to two for tls enabled and tls disabled, so it will give more explicit message for users.